### PR TITLE
adding validation RM108

### PR DESCRIPTION
**Pull request by maimorag**
## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-10610

## Description
This validation verifies the following in README and Description files:
1. Image links are relative.
2. Images are saved under "doc_files" folder.

### CODE REVIEW

File: demisto_sdk/commands/validate/sdk_validation_config.toml [hunk 1]
```diff
@@ -7,7 +7,7 @@ ignorable_errors = [
     "MP106",
     "PA113", "PA116", "PA124", "PA125", "PA127", "PA129",
     "PB104", "PB105", "PB106", "PB107", "PB110", "PB111", "PB114", "PB115", "PB116", "PB118", "PB119", "PB121",
-    "RM100", "RM102", "RM104", "RM106", "RM108", "RM110", "RM112", "RM113",
+    "RM100", "RM102", "RM104", "RM106", "RM110", "RM112", "RM113",
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/sdk_validation_config.toml:10]: Why did you remove it?

File: demisto_sdk/commands/validate/validators/RM_validators/RM108_check_readme_image.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        "Using relative references to files in the repo folder enhances security by reducing reliance"
+        " on external links, minimizing the risk of link manipulation or redirection attacks. "
+    )
+    error_message = (
+        "{}. See https://xsoar.pan.dev/docs/integrations/integration-docs#images for further info on"
+        " how to add images to pack markdown files."
+    )
+    related_field = "readme, description"
+    is_auto_fixable = False
+    expected_git_statuses = [GitStatuses.ADDED, GitStatuses.MODIFIED]
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/RM_validators/RM108_check_readme_image.py:41]: I think we can remove this.

File: demisto_sdk/commands/validate/validators/RM_validators/RM108_check_readme_image.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+            for content_item in content_items
+            if (
+                error_message := self.verify_absolute_images_not_exist(
+                    content_item.readme.file_content
+                )
+                + self.verify_relative_saved_in_doc_files(
+                    content_item.readme.file_content
+                )
+                + self.verify_absolute_images_not_exist(content_item.description)
+                + self.verify_relative_saved_in_doc_files(content_item.description)
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/RM_validators/RM108_check_readme_image.py:60]: did you mean description_file?

File: demisto_sdk/commands/validate/validators/RM_validators/RM108_check_readme_image.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+            )
+        ]
+
+    def verify_absolute_images_not_exist(self, content_item: Optional[str]) -> str:
+        """Check for existing absolute image paths."""
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/RM_validators/RM108_check_readme_image.py:65]: Add args & return sections

File: demisto_sdk/commands/validate/validators/RM_validators/RM108_check_readme_image.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        )
+        if matches:
+            absolute_links = " \n".join(
+                match[1] if match[0] else match[2] for match in matches
+            )
+            return f"Invalid image path(s), use relative paths instead in the following links:\n{absolute_links}.\n"
+        return ""
+
+    def verify_relative_saved_in_doc_files(self, content_item: Optional[str]) -> str:
+        """Check for relative image paths not saved in the pack's doc_files folder."""
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/RM_validators/RM108_check_readme_image.py:79]: Add args & return sections

**Review by YuvHayun — CHANGES_REQUESTED** [review]
Nice! see my comments.

File: demisto_sdk/commands/validate/sdk_validation_config.toml (hunk 1 above)
Comment (maimorag) [reply demisto_sdk/commands/validate/sdk_validation_config.toml:10]: https://jira-dc.paloaltonetworks.com/browse/CIAC-10610, if I understand correctly, it needs to be removed to
fix all places in the content repo where the validation is ignored

File: demisto_sdk/commands/validate/validators/RM_validators/RM108_check_readme_image.py (hunk 2 above)
Comment (maimorag) [reply demisto_sdk/commands/validate/validators/RM_validators/RM108_check_readme_image.py:41]: sure

File: demisto_sdk/commands/validate/sdk_validation_config.toml (hunk 1 above)
Comment (YuvHayun) [inline demisto_sdk/commands/validate/sdk_validation_config.toml:10]: Don't forget to set it in use_git/path_based.select section

File: demisto_sdk/commands/validate/validators/RM_validators/RM108_check_image_path_only_readme.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+            for content_item in content_items
+            if (
+                error_message := self.verify_absolute_images_not_exist(
+                    content_item.readme.file_content
+                )
+                + self.verify_relative_saved_in_doc_files(
+                    content_item.readme.file_content
+                )
+            )
+        ]
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/RM_validators/RM108_check_image_path_only_readme.py:39]: I think we can move it to ImagePathValidator and have ImagePathIntegrationValidator inherit and overwrite the relevant fields.
But up to you to decide.

File: demisto_sdk/commands/validate/tests/RM_validators_test.py [hunk 7]
```diff
@@ -434,3 +440,105 @@ def test_IsContainDemistoWordValidator_is_invalid():
...
+        "\nhttps://www.example.com/images/example_image.jpgRelative image paths found outside the pack's "
+        "doc_files directory. Please move the following images to the doc_files directory:"
+        "\n../../content/image.jpg. See https://xsoar.pan.dev/docs/integrations/integration-docs#images"
+        " for further info on how to add images to pack markdown files."
+    )
+    result = ImagePathIntegrationValidator().is_valid(content_items)
+    assert result[0].message == expected
+
+
+def test_ImagePathOnlyReadMeValidator_is_valid():
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/tests/RM_validators_test.py:496]:
```suggestion
def test_ImagePathOnlyReadMeValidator_is_valid_valid_case():
```

File: demisto_sdk/commands/validate/tests/RM_validators_test.py [hunk 8]
```diff
@@ -434,3 +440,105 @@ def test_IsContainDemistoWordValidator_is_invalid():
...
+    """
+    content_items = [
+        create_integration_object(
+            readme_content="![Example Image](../doc_files/image.png)",
+        ),
+    ]
+    assert not ImagePathOnlyReadMeValidator().is_valid(content_items)
+
+
+def test_ImagePathOnlyReadMeValidator_invalid():
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/tests/RM_validators_test.py:514]:
```suggestion
def test_ImagePathOnlyReadMeValidator_is_valid_invalid_case():
```

File: demisto_sdk/commands/validate/tests/RM_validators_test.py [hunk 9]
```diff
@@ -434,3 +440,105 @@ def test_IsContainDemistoWordValidator_is_invalid():
...
+    content_items = [
+        create_integration_object(
+            readme_content="![Example Image](../doc_files/image.png)",
+            description_content="valid description ![Example Image](../doc_files/image.png)",
+        ),
+    ]
+    assert not ImagePathIntegrationValidator().is_valid(content_items)
+
+
+def test_ImagePathIntegrationValidatorinvalid():
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/tests/RM_validators_test.py:464]:
```suggestion
def test_ImagePathIntegrationValidator_is_valid_invalid_case():
```

File: demisto_sdk/commands/validate/tests/RM_validators_test.py [hunk 10]
```diff
@@ -434,3 +440,105 @@ def test_IsContainDemistoWordValidator_is_invalid():
     results = IsContainDemistoWordValidator().is_valid(content_items)
     assert len(results) == 1
     assert results[0].message == expected_msg
+
+
+def test_ImagePathIntegrationValidator_is_valid():
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/tests/RM_validators_test.py:445]:
```suggestion
def test_ImagePathIntegrationValidator_is_valid_valid_case():
```

**Review by YuvHayun — APPROVED** [review]
Great job!
minor comments.

File: demisto_sdk/commands/validate/tests/RM_validators_test.py [hunk 11]
```diff
@@ -434,3 +440,105 @@ def test_IsContainDemistoWordValidator_is_invalid():
...
+    content_items = [
+        create_integration_object(
+            readme_content=" Readme contains absolute path:\n 'Here is an image:\n"
+            " ![Example Image](https://www.example.com/images/example_image.jpg)",
+            description_content="valid description ![Example Image](../../content/image.jpg)",
+        ),
+    ]
+    expected = (
+        "Invalid image path(s) detected. Please use relative paths instead in the following links:"
+        "\nhttps://www.example.com/images/example_image.jpgRelative image paths found outside the pack's "
```
Comment (eyalpalo) [inline demisto_sdk/commands/validate/tests/RM_validators_test.py:487]: https://www.example.com/images/example_image.jpgRelative lets change to  have a space between the .jpg and the Relative.

File: demisto_sdk/commands/validate/tests/RM_validators_test.py [hunk 12]
```diff
@@ -434,3 +440,105 @@ def test_IsContainDemistoWordValidator_is_invalid():
...
+    content_items = [
+        create_integration_object(
+            readme_content=" Readme contains absolute path:\n 'Here is an image:\n"
+            " ![Example Image](https://www.example.com/images/example_image.jpg)"
+            " ![Example Image](../../content/image.jpg)",
+        ),
+    ]
+    expected = (
+        "Invalid image path(s) detected. Please use relative paths instead in the following links:"
+        "\nhttps://www.example.com/images/example_image.jpgRelative image paths found outside the pack's"
```
Comment (eyalpalo) [inline demisto_sdk/commands/validate/tests/RM_validators_test.py:537]: Same as above.

File: demisto_sdk/commands/validate/validators/RM_validators/RM108_check_image_path.py [hunk 13]
```diff
@@ -0,0 +1,117 @@
...
+            str -- The error message if the content item contains absolute paths.
+        """
+        absolute_links = re.findall(
+            URL_IMAGE_LINK_REGEX + r"|" + HTML_IMAGE_LINK_REGEX,
+            content_item,
+            re.IGNORECASE | re.MULTILINE,
+        )
+        if absolute_links:
+            absolute_links = [
+                absolute_link[1] if absolute_link[0] else absolute_link[2]
```
Comment (eyalpalo) [inline demisto_sdk/commands/validate/validators/RM_validators/RM108_check_image_path.py:76]: What is this meant for?

**Review by eyalpalo — CHANGES_REQUESTED** [review]
Nice!

File: demisto_sdk/commands/validate/validators/RM_validators/RM108_check_image_path.py [hunk 14]
```diff
@@ -0,0 +1,117 @@
...
+            ValidationResult(
+                validator=self,
+                message=self.error_message.format(error_message),
+                content_object=content_item,
+            )
+            for content_item in content_items
+            if (error_message := self.validate_content_items(content_item))
+        ]
+
+    def validate_content_items(self, content_item):
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/RM_validators/RM108_check_image_path.py:49]: Very nice!

**Review by YuvHayun — APPROVED** [review]

File: demisto_sdk/commands/validate/validators/RM_validators/RM108_check_image_path.py (hunk 13 above)
Comment (maimorag) [reply demisto_sdk/commands/validate/validators/RM_validators/RM108_check_image_path.py:76]: searching with the regex returns each link group  ("image]", "link" l)not only with the link itself.  Since MD file has the same structure for images we can use this function to get the link itself

File: demisto_sdk/commands/validate/validators/RM_validators/RM108_check_image_path.py (hunk 14 above)
Comment (maimorag) [reply demisto_sdk/commands/validate/validators/RM_validators/RM108_check_image_path.py:49]: thanks :)

**Review by eyalpalo — APPROVED** [review]
